fix(eslint): support custom pageExtensions in no-html-link-for-pages rule

Amogh-2404 opened the pull request:
## Description

Fixes #53473

The `@next/next/no-html-link-for-pages` ESLint rule previously hardcoded page extensions as `.js`, `.jsx`, `.ts`, `.tsx` and did not respect custom `pageExtensions` configured in `next.config.js`.

This PR adds support for a second optional parameter `pageExtensions` to the rule configuration, allowing developers to specify custom file extensions that should be recognized as pages.

## Changes

- Extended ESLint rule schema to accept optional `pageExtensions` array parameter
- Updated `parseUrlForPages()` to dynamically build regex from extensions
- Updated `parseUrlForAppDir()` to dynamically build regex from extensions  
- Updated exported utility functions with default `pageExtensions` parameter
- Added comprehensive test suite for custom extensions (`.page.tsx`, `.mdx`, etc.)
- Updated documentation with usage examples
- Maintained backward compatibility (defaults to `['tsx', 'ts', 'jsx', 'js']`)
- Fixed unrelated failing test in `web.test.ts` (stream body onClose callback)

## Example Usage

```json
{
  "rules": {
    "@next/next/no-html-link-for-pages": [
      "error",
      "pages/",
      ["page.tsx", "page.ts", "mdx"]
    ]
  }
}
```

## Test Results

✅ All 232 test suites passed
✅ 1,810 tests passed
✅ 266 snapshots passed

## Breaking Changes

None - fully backward compatible

Amogh-2404 committed:
fix(eslint): support custom pageExtensions in no-html-link-for-pages rule

Fixes #53473

The `@next/next/no-html-link-for-pages` ESLint rule previously hardcoded
page extensions as `.js`, `.jsx`, `.ts`, `.tsx` and did not respect custom
`pageExtensions` configured in next.config.js.

This change adds support for a second optional parameter `pageExtensions`
to the rule configuration, allowing developers to specify custom file
extensions that should be recognized as pages.

Changes:
- Extended ESLint rule schema to accept optional pageExtensions array
- Updated parseUrlForPages() to dynamically build regex from extensions
- Updated parseUrlForAppDir() to dynamically build regex from extensions
- Updated exported utility functions with default pageExtensions parameter
- Added comprehensive test suite for custom extensions (.page.tsx, .mdx, etc)
- Updated documentation with usage examples
- Maintained backward compatibility (defaults to ['tsx', 'ts', 'jsx', 'js'])

Example usage:
```json
{
  "rules": {
    "@next/next/no-html-link-for-pages": [
      "error",
      "pages/",
      ["page.tsx", "page.ts", "mdx"]
    ]
  }
}
```

Amogh-2404 committed:
fix(server): ensure onClose callback fires when stream body is consumed

The trackStreamConsumed function wasn't reliably calling the onEnd callback
when response bodies were fully consumed. This was causing the onClose
callback to not fire in certain streaming scenarios.

The issue was that we were only monitoring when the source stream finished
piping, but not when the destination stream was actually consumed.

This fix uses both the TransformStream's flush() method (for clean closes)
and the pipeTo promise handlers (for cancellations/errors) to ensure the
callback is triggered reliably. A guard flag prevents duplicate calls.

Fixes failing test: WebNextResponse onClose > stream body

Amogh-2404 committed:
chore: add .claude to gitignore

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: 0c66a2db22fbb8caa6d75d5cae27c56c406e313e

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: 9b6a0ff8f8faf4db7ef63fdb354299d23ee8122b

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

Amogh-2404 committed:
fix(eslint): remove incorrect parameter from isDirectory() call

The isDirectory() method on Dirent objects doesn't take any parameters.
Fixed incorrect call in parseUrlForAppDir function.

Co-authored-by: graphite-app[bot]

Amogh-2404 committed:
fix: apply prettier formatting to test file
